### PR TITLE
Fix/1207 deleted entries shown in interactive search

### PR DESCRIPTION
**Pull request by deicon**
* Deleted entries have not been excluded from the db queries
* Using deleted_at field to include/exclude deleted items

### CODE REVIEW

File: atuin/src/command/client/history.rs [hunk 1]
```diff
@@ -266,6 +270,7 @@ impl Cmd {
         Ok(())
     }
 
+    #[allow(clippy::too_many_arguments)]
```
Comment (deicon) [inline atuin/src/command/client/history.rs:273]: @ellie @conradludgate Not happy about this one, but seems to be a decision of yours, to not use subcommand structs to pass arguments into methods.

Comment (conradludgate) [reply atuin/src/command/client/history.rs:273]: Not really a thought-out decision, It's just the easiest thing to do over time. This isn't a library so the chances of making mistakes with high arg count functions is low.

Comment (deicon) [reply atuin/src/command/client/history.rs:273]: Totally agree on this one. Having structs with many booleans is also not much better and will also break clippy. So I would consider this a reasonable decision. Even if it not a decision :)

File: atuin/src/command/client/history.rs [hunk 2]
```diff
@@ -51,6 +51,10 @@ pub enum Cmd {
         #[arg(long)]
         cmd_only: bool,
 
+        /// explicitely include deleted
+        #[arg(long)]
+        include_deleted: bool,
```
Comment (ellie) [inline atuin/src/command/client/history.rs:56]: I don't _think_ there is ever a time when the user would want to include deleted records in outputs, so I think we're safe to only "include deleted" from sync functions - elsewhere, just default to false.

**Review by ellie — APPROVED** [review]
thank you! 🙌

**Comment by deicon** [comment]
@ellie reverted your auto merge as cargo fmt was missing. my bad

**Comment by deicon** [comment]
Anything missing still here ?

**Comment by deicon** [comment]
@ellie Alright. Missed the print0 PR as it came later. I added the missing documentation of the print0 parameter as well after fixing the conflict

**Review by ellie — DISMISSED** [review]

**Comment by deicon** [comment]
@ellie I keep forgetting cargo fmt. need to update my config

**Review by ellie — APPROVED** [review]

**Pull request merged**